### PR TITLE
Gated Recurrent Unit for Heart Anomaly Detection

### PR DESCRIPTION
**Pull request by viebboy**
This PR implements the gated recurrent unit learner for the heart anomaly detection task. This PR provides the source code, unit tests, and documentation. The pre-trained models and preprocessed dataset have been uploaded to the FTP server.

Note: the code utilizes the pytorch training functionalities from #70.

### CODE REVIEW

**Comment by passalis** [comment]
@viebboy testing is now available again. So, when you are ready, you can use the "test sources" label to run the style checks and "test tools" to run the tool tests.

**Comment by viebboy** [comment]
@stefaniapedrazzi @passalis This PR is ready for review.

**Review by stefaniapedrazzi — APPROVED** [review]
Thank you
